.github/workflows/test.yml

```diff
@@ -66,4 +66,30 @@ jobs:
         uses: actions/upload-artifact@v2
         with:
           name: code-coverage-report
-          path: cobertura.xml
\ No newline at end of file
+          path: cobertura.xml
+
+  miri:
+    name: Miri UB tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      
+      - name: Set correct nightly version
+        id: nightly-version
+        run: echo "::set-output name=miri-version::nightly-$(curl -s https://rust-lang.github.io/rustup-components-history/x86_64-unknown-linux-gnu/miri)"
+
+      - name: Install ${{ steps.nightly-version.outputs.miri-version }} toolchain
```

Comment:
Add a step somewhere to install xargo so it doesn't have to be reinstalled every time